### PR TITLE
Events page

### PR DESCRIPTION
**Pull request by kulloveth**
events
events_dark
home_dark
new_home

### CODE REVIEW

File: app/src/main/java/com/madonasyombua/sportsdb/app/SportsApp.kt [hunk 1]
```diff
@@ -2,12 +2,26 @@ package com.madonasyombua.sportsdb.app
...
 import com.madonasyombua.sportsdb.ui.screen.home.HomeScreen
+import com.madonasyombua.sportsdb.ui.screen.teamdetails.TeamDetailsScreen
 
 @ExperimentalFoundationApi
 @Composable
-fun TheSportsApp() {
-    HomeScreen()
+fun TheSportsApp(toggle:()->Unit) {
+    val navController = rememberNavController()
+    NavHost(navController, startDestination = "home") {
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/app/SportsApp.kt:17]: This is amazing 😮

File: app/src/main/java/com/madonasyombua/sportsdb/data/remote/api/Api.kt [hunk 2]
```diff
@@ -17,15 +15,27 @@ import retrofit2.http.Query
  */
 interface Api {
     @GET("searchteams.php")
-    fun getAllTeams(@Query("t") team : String) : Single<SportsBaseResponse>
+    fun getAllTeams(@Query("t") team: String): Single<SportsBaseResponse>
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/data/remote/api/Api.kt:18]: 👏🏽

File: app/src/main/java/com/madonasyombua/sportsdb/data/remote/response/LeagueTableResponse.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.madonasyombua.sportsdb.data.remote.response
+
+import com.madonasyombua.sportsdb.data.remote.model.LeagueTable
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class LeagueTableResponse(
+    val table:List<LeagueTable>
+)
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/data/remote/response/LeagueTableResponse.kt:9]: Do we need to `serialize` this?

File: app/src/main/java/com/madonasyombua/sportsdb/ui/MainActivity.kt [hunk 4]
```diff
@@ -4,19 +4,29 @@ import android.os.Bundle
...
 @AndroidEntryPoint
 class MainActivity : AppCompatActivity() {
+
     @ExperimentalFoundationApi
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            MaterialTheme {
-                TheSportsApp()
+            val darkTheme = remember { mutableStateOf(false) }
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/MainActivity.kt:22]: Great add 👏🏽

File: app/src/main/java/com/madonasyombua/sportsdb/ui/components/AppBar.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import androidx.compose.material.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+
+/**
+ * Created by Loveth Nwokike
+ * 3/22/2021
+ * */
+@Composable
+fun AppBar(title: String, actions: @Composable RowScope.() -> Unit = {}) {
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/components/AppBar.kt:14]: I am kind of wondering how one would  declare the full screen flags on` compose `

File: app/src/main/java/com/madonasyombua/sportsdb/ui/screen/home/HomeScreen.kt [hunk 6]
```diff
@@ -1,93 +1,108 @@
...
-                        tint = Color.White
-                    )
-                }
+fun HomeAppBar(toogle:()->Unit) {
+    AppBar(title = "Sports App", actions = {
+
+        CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
+            IconButton(
+                onClick = toogle
+
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/screen/home/HomeScreen.kt:88]: Extra space 👾

File: app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/TeamDetails.kt [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+@Composable
+fun TeamDetailsScreen(id:String) {
+    val viewModel : TeamEventViewModel = hiltNavGraphViewModel()
+    viewModel.getHomeTeamDetails(id)
+    val teamDetails = viewModel.teamDetailsLiveData.observeAsState()
+    val navController = rememberNavController()
+    Scaffold(topBar= { AppBar(title = "",actions =
+    {
+        Row(modifier = Modifier.fillMaxWidth(),horizontalArrangement = Arrangement.Center) {
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/TeamDetails.kt:48]: I'm jealous they have Row 😱

File: app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/event/TeamLastEvents.kt [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    ) {
+        events.value?.let {events->
+            items(events.size) { index ->
+                    EventsCardItem(event = events[index], viewModel)
+
+            }}
+    }
+
+}
+
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/event/TeamLastEvents.kt:50]: 👾

File: app/src/main/java/com/madonasyombua/sportsdb/ui/theme/Type.kt [hunk 9]
```diff
@@ -1,2 +1,69 @@
...
+import com.madonasyombua.sportsdb.R
+
+
+private val Montserrat = FontFamily(
+    Font(R.font.montserrat_regular),
+    Font(R.font.montserrat_medium, FontWeight.W500),
+    Font(R.font.montserrat_semibold, FontWeight.W600)
+)
+
+val SportsAppTypography = Typography(
```
Comment (Madonahs) [inline app/src/main/java/com/madonasyombua/sportsdb/ui/theme/Type.kt:19]: 😮

**Review by Madonahs — APPROVED** [review]
This is great work 👍🏽  thank you for taking your time and dedication

File: app/src/main/java/com/madonasyombua/sportsdb/data/remote/response/LeagueTableResponse.kt (hunk 3 above)
Comment (kulloveth) [reply app/src/main/java/com/madonasyombua/sportsdb/data/remote/response/LeagueTableResponse.kt:9]: not necessary only integer is been passed around

File: app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/TeamDetails.kt (hunk 7 above)
Comment (kulloveth) [reply app/src/main/java/com/madonasyombua/sportsdb/ui/screen/teamdetails/TeamDetails.kt:48]: yup can wait for compose to be stable. android will become easier

**Pull request merged**